### ISSUE TITLE
bootstage2: RP2040 only programmable after flashing once with PICO_BOOT_STAGE2_CHOOSE_IS25LP080 1 defined

### ISSUE THREAD

**Issue by fabianhugo** [raspberrypi/pico-sdk#785]
Dear everyone, 
I have a custom board with RP2040B2 and ISSI IS25LP128F flash in QSPI configuration. First some good news, the combination of this flash and the RP2040 works almost right away and I have succesfully flashed the pico-example _blink_. So if there is anywhere a list with supported flash ICs for the RP2040, i think this flash IC should be added. 
But a thing that keeps on puzzling me is, that my boards when they come from the factory can not be successfully programmed right away via Bootloader Mode. I built the blink example in standard configuration, the only thing I changed is in blink.c the led pin. So it is built for pico boards and #define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1. After copying the generated UF2 to the board, it restarts, isn't shown in file explorer anymore but does not blink the LED. 
And if I flash the blink example generated with #define PICO_BOOT_STAGE2_CHOOSE_IS25LP080 1 to the board, the LED still stays off. But then if I flash any new progam to it, e.g. the blink example with #define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1 the LED starts to  blink. So I think the RP2040 is compatible with my flash using the standard PICO_BOOT_STAGE2_CHOOSE_W25Q080 option because then also every other UF2 is successfully programmed to the board then. But with every new board I still need to at least once flash a #define PICO_BOOT_STAGE2_CHOOSE_IS25LP080 first. 
I also tried nuking the flash first and then do a regular PICO_BOOT_STAGE2_CHOOSE_W25Q080 example but that fails as well. 

Has anyone an idea or pointer where I should look? 

PS: My setup is WSL2 Ubuntu on a Win10 machine, pico sdk and pico-examples are up to date. I used Picotool to examine the generated files but could not find any hint.

**Comment by fabianhugo** [comment]
Below is the board file header _mini21.h_ that I use to build the examples. To build I run:  
1. makedir build && cd build 
2. cmake -D"PICO_BOARD=MINI21" -DCMAKE_BUILD_TYPE=DEBUG .. 
3. cd blink
4. make
```
/*
 * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

// -----------------------------------------------------
// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
// -----------------------------------------------------


// For board detection
#define MINI21

// --- UART ---
#ifndef PICO_DEFAULT_UART
#define PICO_DEFAULT_UART 1
#endif
#ifndef PICO_DEFAULT_UART_TX_PIN
#define PICO_DEFAULT_UART_TX_PIN 20
#endif
#ifndef PICO_DEFAULT_UART_RX_PIN
#define PICO_DEFAULT_UART_RX_PIN 21
#endif

// --- LED ---
#ifndef PICO_DEFAULT_LED_PIN
#define PICO_DEFAULT_LED_PIN 28
#endif
// no PICO_DEFAULT_WS2812_PIN


// --- FLASH ---

#define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1

#ifndef PICO_FLASH_SPI_CLKDIV
#define PICO_FLASH_SPI_CLKDIV 2
#endif

#ifndef PICO_FLASH_SIZE_BYTES
#define PICO_FLASH_SIZE_BYTES (16 * 1024 * 1024)
#endif

// All boards have B1 RP2040
#ifndef PICO_RP2040_B0_SUPPORTED
#define PICO_RP2040_B0_SUPPORTED 0
#endif

```

**Comment by fabianhugo** [comment]
Update: 
I assume this problem was caused by already configured flash parts, probably the QE bit or some other non-volatile flash configuration was responsible. I know that the Arduino Connect also works with with IS25LP128F flash using the pico standard W25Q080 driver. 
I tried manually configuring the flash using JLink Flash SPI and connecting a JLink in SPI mode, but failed to reproduce this state. Anyhow, I think this shouldnt occur with newly bought flash pieces, therefore I close this issue now.
